enhancements/sig-policy/130-policy-merge-compliancetypes/README.md

```diff
@@ -0,0 +1,376 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+We propose to add new options for the `spec["object-templates][*].complianceType` field in
+ConfigurationPolicies to allow the user to specify new merge strategies: `musthaveapply`, 
+`musthavestrategic` and `musthavemerge`. These new options correspond to existing options in
+`kubectl`, which users may be familiar with. Along with the existing `musthave`, `mustonlyhave`, and
+`mustnothave` options, this should provide more flexibility and control to users, and be able to
+provide a more predictable experience.
+
+## Motivation
+
+The existing `musthave` complianceType has a bespoke algorithm for merging content from the cluster
+state with what is defined in the policy. One specific difference is when the policy includes a list
+in the object template: `musthave` will check the existing list for any items specified in the
+policy, and if any are missing, it will *add* them to the list and apply the change. Tools like
+`kubectl apply` can do something similar on some objects, via a "strategic merge", but not on custom
+resources.
+
+The existing `mustonlyhave` complianceType is simpler to understand: any fields on the object not
+defined in the policy will be removed from the object on the cluster (possibly just reverting them
+to default values). This can be used to avoid other merge behaviors, but requires the policy author
+to include more of the object in the policy, and does not allow the unset fields to be set by other
+controllers. This complianceType has been known to cause issues by repeatedly updating objects on
+the cluster, causing high API server load.
+
+New complianceTypes would give more tools to policy authors to specify the exact behavior they want.
+
+### Goals
+
+1. Policy authors should be able to create a policy to *replace* a specific list in an object on the
+   cluster with what is defined in the policy, *without* that policy reverting other customizations
+   on the object.
+2. Policy authors should be able to create a policy that has the exact same behavior as `kubectl`
+   commands like `apply`, `patch --type=strategic` and `patch --type=merge`.
+3. Clarify in documentation the exact behaviors of the complianceType options.
+
+### Non-Goals
+
+1. Support strategic merge on custom resources (`kubectl` does not support this).
+
+## Proposal
+
+Add a `musthavemerge` complianceType, corresponding to the `kubectl` command `patch --type=merge`.
+In particular, this allows for replacement of specific fields / objects / lists in an object without
+reverting other "unrelated" changes in the object. This merge strategy works on all resource types,
+including custom ones.
+
+Add a `musthavestrategic` complianceType, corresponding to `kubectl patch --type=strategic`. This
```

Comment:
What would happen if the user specified this and it was applied on a custom resource? Would it fall back to merge?

Reply:
Something like `kubectl patch operatorpolicy --type=strategic ...` fails with an error mentioning that `application/strategic-merge-patch+json is not supported`.